pdf/pdfrenderthread.cpp

```diff
@@ -33,6 +33,72 @@ class PDFRenderThreadQueue;
 
 const QEvent::Type Event_JobPending = QEvent::Type(QEvent::User + 1);
 
+class SearchThread: public QThread
+{
+    Q_OBJECT
+public:
+    SearchThread(Poppler::Document *document, QObject *parent = 0)
+        : QThread(parent), m_document(document)
+    {
+    }
+    ~SearchThread()
+    {
+        requestInterruption();
+        wait();
+    }
+
+    QList<QPair<int, QRectF>> m_matches;
+
+    void start(const QString& search, uint startPage = 0)
```

Comment:
Well, I've named it on purpose, because it has the meaning of starting the work thread from the main one like QThread::start() is doing but doing here for the specialised SearchThread class, so with additional arguments.

Besides, if you think it's misleading as your initial remark implies, I'll change it to startSearch() without any trouble ;)